### PR TITLE
Feature/apollo client setup

### PR DESCRIPTION
**Pull request by LKessell**
# 

- [x] New feature 
- [ ] Refactor
- [ ] Bugfix 
- [ ] Documentation

## About

Set up the Apollo Client to allow communication with the deployed back-end server

## Details

Update dependencies
Add the Apollo Client
Add GraphQL query to the Character page to pull data from the back-end
Update CharacterPage component to use queried data (Name, Species, Career, Specialization)

## Concerns
Will need to update other character information to use query data once available

### CODE REVIEW

File: src/components/CharacterPage.js [hunk 1]
```diff
@@ -1,23 +1,41 @@
 import React from "react";
 import { InfoField } from "./InfoField";
 import { Characteristics } from "./Characteristics";
+import { useQuery, gql } from "@apollo/client";
 import { sampleData } from "../mock-data";
 
+const CHARACTER = gql`
```
Comment (andrewvallejo) [inline src/components/CharacterPage.js:7]: This is cool. its like prop types, fetch calls, and actual value all in one

File: src/components/CharacterPage.js [hunk 2]
```diff
@@ -1,23 +1,41 @@
...
+  if (loading) return 'Loading your data...';
+  if (error) return `Error! ${error.message}`;
+
   const characterInfo = (
     <>
-    <InfoField heading={Object.keys(character)[0]} info={character.name}/>
-    <InfoField heading={Object.keys(character)[1]} info={character.species}/>
-    <InfoField heading={Object.keys(character)[2]} info={character.career}/>
-    <InfoField heading={Object.keys(character)[3]} info={character.specialization}/>
+    <InfoField heading={Object.keys(data.character)[3]} info={data.character.name}/>
```
Comment (andrewvallejo) [inline src/components/CharacterPage.js:28]: I'm curious about this method of interpolating that data. Could these be accessed by using the actual key?

**Review by andrewvallejo — APPROVED** [review]

**Comment by andrewvallejo** [comment]
I left you a question! Other wise this looks great! I'm excited to plug more of it in!

File: src/components/CharacterPage.js (hunk 2 above)
Comment (LKessell) [reply src/components/CharacterPage.js:28]: Yeah, I don't feel like this is an optimal solution. I'd love to circle back to this and discuss once we have more data to work with!

**Pull request merged**